src/gcp/cloudfunctions.ts

```diff
@@ -559,6 +560,9 @@ export function endpointFromFunction(gcfFunction: CloudFunction): backend.Endpoi
     );
   }
   endpoint.codebase = gcfFunction.labels?.[CODEBASE_LABEL] || projectConfig.DEFAULT_CODEBASE;
+  if (gcfFunction.labels?.[HASH_LABEL]) {
+    endpoint.hash = gcfFunction.labels?.[HASH_LABEL];
```

Comment:
nit* since you already checked for existence of `HASH_LABEL`, I think you can omit the `?` (think TS is smart enough to deduce that).

Reply:
Yes, any time you're in the clause of an if statement that did a null check, the compiler stops requiring the ?. And in fact we should remove it to
A) Prove to ourselves that we've handled all edge cases
B) Ensure that this code, if copy + pasted, will only work when similarly guarded

Reply:
Good catch! Updated!